endpoints/remotehost/remotehost

```diff
@@ -199,17 +201,13 @@ function cleanup_osruntime() {
 
         if [ -e $endpoint_run_dir/chroot-container-mount.txt ]; then
             container_mount=`cat $endpoint_run_dir/chroot-container-mount.txt`
-            for fs in proc dev dev/hugepages proc sys lib/modules usr/src boot; do
-                if [ -e $container_mount/$fs ]; then
-                    echo "Removing [$container_mount/$fs] for chroot using container [$container_name]"
-                    do_ssh $user@$host umount $container_mount/$fs
-                else
-                    echo "NOT removing [$container_mount/$fs] for chroot using container [$container_name] because it does not appear to be mounted"
-                    do_ssh $user@$host mount | grep $container_mount/$fs
-                fi
+            for fs in ${chroot_rbind_mounts}; do
+                echo "Removing ${fs} from ${container_mount}"
+                do_ssh $user@$host umount --verbose --recursive ${container_mount}/${fs}
```

Comment:
When this line runs I get no output, even thought it has `--verbose` and I see output from other `do_ssh` invocations.  Any idea why that would be?  This caused issues discussed below...

Reply:
does umount only use stderr and maybe that is not output by ssh?

Reply:
I was manually able to run successful mount/umount commands through a ssh call without issue:

Reply:
More examples:

```
$ ssh root@<ip> mount --verbose -o rbind /proc /tmp/proc
mount: /proc bound on /tmp/proc.

$ ssh root@<ip> mount --verbose --make-rslave /tmp/proc
mount: /tmp/proc propagation flags changed.

$ ssh root@<ip> umount --verbose --recursive /tmp/proc
umount: /tmp/proc/sys/fs/binfmt_misc unmounted
umount: /tmp/proc/sys/fs/binfmt_misc (systemd-1) unmounted
umount: /tmp/proc unmounted
```

Reply:
When this code runs it dumps this on setup:

```
Adding host directories to container mount
Adding proc to /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged
mount: /proc bound on /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged/proc.
mount: /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged/proc propagation flags changed.

Adding dev to /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged
mount: /dev bound on /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged/dev.
mount: /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged/dev propagation flags changed.

Adding sys to /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged
mount: /sys bound on /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged/sys.
mount: /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged/sys propagation flags changed.

Adding lib/modules to /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged
mount: /usr/lib/modules bound on /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged/usr/lib/modules.
mount: /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged/usr/lib/modules propagation flags changed.

Adding usr/src to /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged
mount: /usr/src bound on /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged/usr/src.
mount: /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged/usr/src propagation flags changed.

Adding boot to /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged
mount: /boot bound on /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged/boot.
mount: /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged/boot propagation flags changed.
```
but does this on cleanup:

```
Removing proc from /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged

Removing dev from /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged

Removing sys from /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged

Removing lib/modules from /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged

Removing usr/src from /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged

Removing boot from /var/lib/containers/storage/overlay/1e97b59abe52974ae3b644db08d44c0e169b348170ad7e91c6001f950bc8f8f7/merged
```

Reply:
I did a test inside a function, I see the output for success and for failure. Not sure what it could be, if a redirection would cause this.